pkg/apis/v1/nodepool_conversion.go

```diff
@@ -123,9 +123,9 @@ func (in *NodePool) ConvertFrom(ctx context.Context, v1beta1np apis.Convertible)
 		return err
 	}
 	if kubeletAnnotation == "" {
-		in.Annotations = lo.OmitByKeys(in.Annotations, []string{KubeletCompatibilityAnnotationKey})
+		delete(in.Annotations, KubeletCompatibilityAnnotationKey)
 	} else {
-		in.Annotations = lo.Assign(in.Annotations, map[string]string{KubeletCompatibilityAnnotationKey: kubeletAnnotation})
+		in.Annotations = lo.Assign(in.Spec.Template.Annotations, map[string]string{KubeletCompatibilityAnnotationKey: kubeletAnnotation})
```

Comment:
If we are going to add the kubelet annotation to the nodeclaim template spec, we should update this here or else I think this will not work as expected:
- https://github.com/kubernetes-sigs/karpenter/pull/1667/files#diff-b8d4e10f084dfecb98330acb7dc0b082455afea1837c74bd59ab11d29b8518adR126
- https://github.com/kubernetes-sigs/karpenter/pull/1667/files#diff-b8d4e10f084dfecb98330acb7dc0b082455afea1837c74bd59ab11d29b8518adR42

Reply:
I thought about this but figured that it would be more trouble than it's worth. Basically, it would mean that a user could have this annotation on either of two places -- and we would have to reason about both of them to not make a breaking change.

I don't think anything is wrong with the way that this is currently implemented. Can you describe how you are thinking the current configuration is going to cause problems?